File: IMS/IMS.Presentation/DTOs/GetDTO/UserDTO.cs
using Shared.Enums;

namespace IMS.Presentation.DTOs.GetDTO;

public class UserDto
{
    public Guid Id { get; init; }
    public required string Email { get; init; } 
    public required string PhoneNumber { get; init; } 
    public required string FirstName { get; init; } 
    public required string LastName { get; init; } 
    public string? Patronymic { get; init; } 
    public Role Role { get; init; } = Role.Unassigned;
}

File: Frontend/ims-frontend/src/entities/ims/dto/user_dto.ts
import { Role } from '../enums';

export interface CreateUserDto {
    email?: string;
    phoneNumber?: string;
    firstName?: string;
    lastName?: string;
    patronymic?: string;
    role: Role;
}

export interface UserDto {
    id: string;
    email?: string | null;
    phoneNumber?: string | null;
    firstName?: string | null;
    lastName?: string | null;
    patronymic?: string | null;
    role: Role;
}

export interface UpdateUserDto {
    email?: string | null;
    phoneNumber?: string | null;
    firstName?: string | null;
    lastName?: string | null;
    patronymic?: string | null;
    role?: Role;
}

export interface UserDtoPagedList {
    items: UserDto[] | null;
    pageNumber: number;
    pageSize: number;
    totalCount: number;
    hasPrevious: boolean;
    hasNext: boolean;
}
